### DISCUSSION TITLE
Mysql / Mariadb Engine

### DISCUSSION THREAD

**Discussion by morex** [reflex-dev/reflex#1526] [Ideas]
On top of PostGre and sqlite, it would be nice

**Comment by Alek99** [comment]
You should be able to connect and SQL db just by putting in your specifics in the rxconfig.py, lmk if these don't work

**Comment by masenf** [comment]
Any database supported by SQLAlchemy can be used with reflex. [Examples](https://docs.sqlalchemy.org/en/14/core/engines.html#backend-specific-urls) from their docs.

To run a reflex app against mysql database, the following code may be used.

# Run the database server
The following in an insecure example 😉 
```console
docker run -it -p 3306:3306 -e MYSQL_ROOT_PASSWORD=secret -e MYSQL_DATABASE=reflex mysql
```

# Export the DB connection string

The `db_url` may also be set inside `rxconfig.py` file for more permanent configuration.

```console
export DB_URL=mysql+pymysql://root:secret@localhost/reflex
```

# Install database driver

Note that reflex does not include database drivers by default, you must install the appropriate driver yourself.

```console
pip install pymysql cryptography
```

# Create schema

Using `reflex db migrate` will create the appropriate schema on the target database

```console
reflex db migrate
```

# Run the App

```console
reflex run
```
